spring-web/src/main/java/org/springframework/http/HttpHeaders.java

```diff
@@ -682,6 +683,41 @@ public void setContentDispositionFormData(String name, String filename) {
 	}
 
 	/**
+	 * Return the filename parameter of the {@code Content-Disposition} header.
+	 * Also supports {@code filename*} parameter encoded according to RFC 5987
+	 * (only the US-ASCII, UTF-8 and ISO-8859-1 charsets are supported).
+	 * {@code null} is returned if such parameter is not defined.
+	 * @see #setContentDispositionFormData(String, String)
+	 * @see #setContentDispositionFormData(String, String, Charset)
+	 * @since 5.0
+	 */
+	public String getContentDispositionFilename() {
+		String contentDisposition = getFirst(CONTENT_DISPOSITION);
+		if (contentDisposition != null) {
+			String[] parts = StringUtils.tokenizeToStringArray(contentDisposition, ";");
+			if (parts.length > 1) {
+				for (int i = 1; i < parts.length; i++) {
+					String parameter = parts[i];
+					int eqIndex = parameter.indexOf('=');
+					if (eqIndex != -1) {
+						String attribute = parameter.substring(0, eqIndex);
+						String value = (parameter.startsWith("\"", eqIndex + 1) && parameter.endsWith("\"") ?
+								parameter.substring(eqIndex + 2, parameter.length() - 1) :
+								parameter.substring(eqIndex + 1, parameter.length()));
+						if (attribute.equals("filename") ) {
+							return value;
+						}
+						else if (attribute.equals("filename*") ) {
+							return decodeHeaderFieldParam(value);
+						}
+					}
+				}
+			}
+		}
+		return null;
+	}
```

Comment:
It seems Java 8 stream API is not designed to collect successive pair, so I have kept the array based implementation for now.